### PR TITLE
[smaenergymeter] Fix `IllegalStateException`

### PR DESCRIPTION
**Pull request by lsiepel**
- Fixes : https://github.com/openhab/openhab-addons/issues/17280
- Added null annotations
- Replaced some Multicast deprecated methods
- Fix SAT warnings

Test jar 4.2.1+: https://1drv.ms/u/s!AnMcxmvEeupwjv9m9osRyk-2jY9whA?e=ICEKNF
Confirmed to be a working fix, thanks @Hoerli1337

### CODE REVIEW

File: bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java [hunk 1]
```diff
@@ -73,8 +76,8 @@ public void initialize() {
         EnergyMeterConfig config = getConfigAs(EnergyMeterConfig.class);
 
         try {
-            serialNumber = config.getSerialNumber();
-            if (serialNumber == null) {
+            serialNumber = Objects.requireNonNullElse(config.getSerialNumber(), "");
```
Comment (lolodomo) [inline bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java:79]: Why do you need to use Objects.requireNonNullElse while now config.getSerialNumber() returns a non null value ?

Comment (lsiepel) [reply bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java:79]: `config.getSerialNumber()` is `@Nullable`

Comment (lolodomo) [reply bundles/org.openhab.binding.smaenergymeter/src/main/java/org/openhab/binding/smaenergymeter/internal/handler/SMAEnergyMeterHandler.java:79]: Correct, I thought it was what you changed but in fact you kept a nullable value in the config class while you made it non nullable in the other classes.

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Pull request merged**